페어랑 할때만큼은 깔끔하게 짜는 누누입니다 step1

be-student opened the pull request.

woowahan-neo committed:
feat: remove web dependency

woowahan-neo committed:
feat: add example for functional programming api example

woowahan-neo committed:
chore: change package

BGuga committed:
study : jdbc 실습

BGuga committed:
docs: 기능 목록 장성

BGuga committed:
feat: Piece 구현

BGuga committed:
docs : 기능 목록 수정

BGuga committed:
feat: piece 의 기본 생성자 구현

BGuga committed:
feat: space 구현

BGuga committed:
refactor: 도메인 구조 변경 및 기능목록 최신화

BGuga committed:
feat: position 생성

BGuga committed:
feat: Board 구현

BGuga committed:
feat: 체스판 출력 기능 구현

BGuga committed:
feat: 시작, 종료 입력 구현

BGuga committed:
docs : 추가 기능 작성

BGuga committed:
feat: pawnState 구현

BGuga committed:
feat: knightState 구현

BGuga committed:
feat: bishopState 구현

BGuga committed:
feat: rookState 구현

BGuga committed:
feat: queenState 구현

BGuga committed:
feat: kingState 구현

BGuga committed:
feat: position rank, file 차이 반환 기능 구현

BGuga committed:
feat: color 에서 방향 결정하는 기능 구현

BGuga committed:
feat: position 의 직선 경로 반환 기능 구현

BGuga committed:
feat: board 의 말 이동 기능 구현

BGuga committed:
feat: controller 커맨드 기능 구현

BGuga committed:
feat: 출력 양식 변경

BGuga committed:
feat: controller 작성 및 2단계 요구사항 완료

BGuga committed:
refactor: file, rank 경로 생성 메서드 개선

BGuga committed:
refactor: 안 쓰는 메서드 삭제

BGuga committed:
refactor: state 메서드 분리

BGuga committed:
refactor: state 들을 캐싱한다.

BGuga committed:
chore: 잡는 예외 변경 및 출력 양식 변경

BGuga committed:
refactor: state 메서드 가독성 개선

BGuga committed:
refactor: 보드 빈 칸에 빈 피스를 넣어둔다

BGuga committed:
refactor: file 과 rank 의 필드명 변경

BGuga committed:
chore: 메서드 명 통일

BGuga committed:
refactor: 기물 순서 반환 메서드 분리

BGuga committed:
refactor: 메서드 명 변경 및 순서 정리

BGuga committed:
refactor: 컨트롤러 메서드 개선

BGuga committed:
fix: board 이동 버그 수정

be-student committed:
feat: stream 람다 미니 미션 구현

be-student committed:
feat: default 메서드로 공통 로직 분리

be-student committed:
chore: 줄바꿈 수정

be-student committed:
test: board 테스트 케이스 추가

be-student commented on src/main/java/chess/controller/ChessController.java:
```diff
@@ -0,0 +1,97 @@
+    public static final int FILE_INDEX = 0;
+    public static final int RANK_INDEX = 1;
+    public static final int ORIGIN_INDEX = 1;
+    public static final int DEST_INDEX = 2;
```
이 부분 잘못 된 것 같네요

be-student committed:
chore: public 키워드들 제거

drunkenhw commented on src/main/java/chess/domain/dto/PieceResponse.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    public static PieceResponse from(Piece piece) {
+        return new PieceResponse(piece.getType().getType(), piece.getColor().name());
```
`getType().getType()` ? that's coool!

be-student replied:
oh I forgot nonono... too sad...

drunkenhw commented on src/main/java/chess/domain/exception/IllegalPieceMoveException.java:
```diff
@@ -0,0 +1,8 @@
+public class IllegalPieceMoveException extends RuntimeException {
+
+    public IllegalPieceMoveException() {
+        super("잘못 된 기물 움직임 요청입니다.");
```
맞춤법이 잘못 된? 것 같네요

be-student replied:
잘못 된 요청입 니다 죄송 합 니다

drunkenhw commented on src/main/java/chess/domain/state/RookState.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Override
+    public boolean canMove(int x, int y, ColorCompareResult colorCompareResult) {
```
원시값 포장 어떠세요?

be-student replied:
이 부분은 x 값의 변화량, y 값의 변화량, 을 주는건데, 변수명이 잘못된 느낌도 들고
수정해볼만한 느낌이 드네요 좋은데요?

drunkenhw commented on src/main/java/chess/domain/Board.java:
```diff
@@ -0,0 +1,104 @@
+            for (Rank rank : Rank.values()) {
+                piecePosition.computeIfAbsent(Position.of(file, rank), ignored -> Piece.empty());
+            }
+        }
```
2depth that's cooool!

be-student replied:
it's really great
이 부분은 솔직히 분리할만한 가치를 전혀 못 느꼈달까?

drunkenhw commented on src/main/java/chess/domain/Board.java:
```diff
@@ -0,0 +1,104 @@
+        }
+    }
+
+    public List<List<PieceResponse>> getPiecePosition() {
```
dto를 여기서 반환하는 이유가 있나요?

be-student replied:
깊은 복사같은 느낌으로 반환해주는 역할을 해주는 역할을 생각해보면, 도메인쪽에 있는 것이 좋아보였어요

be-student replied:
지금은 view 에 dto 로 넘겼습니다

drunkenhw commented on src/main/java/chess/domain/ColorCompareResult.java:
```diff
@@ -0,0 +1,17 @@
+    DIFFERENT_COLOR,
+    EMPTY;
+
+    public static ColorCompareResult of(Color color1, Color color2) {
```
oh shit `color1, color2`

be-student replied:
color 1,2 를 안 쓰고, 좋은 네이밍을 할 필요성을 못 느꼈달까?

Choi-JJunho replied:
허브 : 이 메소드 자체가 왜 필요한지 모르겠습니다
깃짱 : 이건좀;; 메모리낭비

drunkenhw commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+        return createCrossPath(destination);
+    }
+
+    private void validateStraight(Position destination) {
+        int rankDifference = getRankDifference(destination);
+        int fileDifference = getFileDifference(destination);
+        if (rankDifference == 0 || fileDifference == 0) {
+            return;
+        }
+        if (Math.abs(rankDifference) == Math.abs(fileDifference)) {
+            return;
+        }
+        throw new IllegalPieceMoveException();
+    }
```
두개로 빼면 좋을 것 같은데? - 베로 -

be-student replied:
2개로 뺀다? 2개로 뭘 빼죠?
-누누-

drunkenhw commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+    private List<Position> createDiagonalPath(Position destination) {
+        List<Rank> ranks = rank.createPath(destination.rank);
+        List<File> files = file.createPath(destination.file);
+        ArrayList<Position> result = new ArrayList<>();
```
왜 구체클래스를 -베로-

be-student replied:
제가 한거 아닌데요? 라고 할 예정

drunkenhw commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+    private List<Position> createFilePath(Position destination) {
+        List<File> files = file.createPath(destination.file);
+        return files.stream()
+                .map(it -> Position.of(it, rank))
+                .collect(Collectors.toList());
+    }
+
+    private List<Position> createRankPath(Position destination) {
+        List<Rank> ranks = rank.createPath(destination.rank);
+        return ranks.stream()
+                .map(it -> Position.of(file, it))
```
it을 file이나 rank로 써주면 좋지 않을까요 - 포이 -

be-student replied:
ranks 를 필터링 하는 과정에서 rank 가 필터링 된다는게 너무 명확하기에 
이런 명확한 부분에서는 it 으로 필터링 하는 것이 오히려 직관적이라고 생각했어요
실제로 kotlin 같은 언어에서는 이렇게 쓰인다고 하더라고요

drunkenhw commented on src/main/java/chess/domain/dto/PieceResponse.java:
```diff
@@ -0,0 +1,34 @@
+public class PieceResponse {
+
+    private final String pieceType;
+    private final String pieceColor;
```
pieceColor가 굳이 String일 이유가 있나요

drunkenhw commented on src/main/java/chess/view/OutputView.java:
```diff
@@ -0,0 +1,47 @@
+        System.out.println(e.getMessage());
+    }
+
+    public void printInitialMessage() {
+        System.out.println("> 체스 게임을 시작합니다.");
+        System.out.println("> 게임 시작 : start");
+        System.out.println("> 게임 종료 : end");
+        System.out.println("> 게임 이동 : move source위치 target위치 - 예. move b2 b3");
+    }
```
이 친구 위로 올려주면 안되냐?

be-student replied:
위로 쫌 올리긴 해야겠네 ㅋㅋㅋ

drunkenhw commented on src/test/java/chess/domain/state/KnightStateTest.java:
```diff
@@ -0,0 +1,40 @@
+
+    @ParameterizedTest
+    @CsvSource(value = {"2, 2", "2, -2", "0, 2"})
+    void 나이트_이동_예외_테스트(int xChange, int yChange) {
```
예외 테스트인데 예외가 발생안하는데요?

be-student replied:
앗 예외로 만들었다가 수정해서 예외가 발생하지 않네요 ㅋㅋㅋㅋ
그래도 이동 불가능 상황이라고 생각할만하지 않을까요?

drunkenhw commented on src/test/java/chess/domain/state/PawnStateTest.java:
```diff
@@ -0,0 +1,72 @@
+
+    @ParameterizedTest
+    @ValueSource(strings = {"EMPTY", "SAME_COLOR"})
+    void 폰_대각선_움직임_색_예외_테스트(ColorCompareResult colorCompareResult) {
```
ditto

be-student replied:
ditto

drunkenhw commented on src/main/java/chess/controller/GameCommand.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.regex.Pattern;
+
+public enum GameCommand {
+    START(Pattern.compile("^start$")),
+    END(Pattern.compile("^end$")),
+    MOVE(Pattern.compile("^move [a-h][1-8] [a-h][1-8]$"));
```
that's cooooool - VERO -

be-student replied:
👍 🥇 good

woo-chang replied:
Awesome~!

Choi-JJunho replied:
Amazing

drunkenhw commented on src/main/java/chess/domain/state/MoveState.java:
```diff
@@ -0,0 +1,16 @@
+
+public interface MoveState {
+
+    boolean canMove(int x, int y, ColorCompareResult colorCompareResult);
```
that's not cool - vero -
int x,y 로 한 이유가 있나욥?

be-student replied:
int x,int y 로 한 이유는 x와 y 의 변화량인데요
그걸 전혀 담고 있지 못하네요 인정합니다
죄송합니다

drunkenhw commented on src/test/java/chess/domain/PieceTest.java:
```diff
@@ -0,0 +1,38 @@
+class PieceTest {
+
+    @Test
+    void 피스_이름_가져오기_테스트() {
```
피스 이름 가져오기 테스트? 뭐죱?

be-student replied:
그렇네요?
저게 뭐지...? 제가 안 짰어요

drunkenhw commented on src/test/java/chess/domain/BoardTest.java:
```diff
@@ -0,0 +1,88 @@
+    @Test
+    void 두_뻔쨰_줄_테스트() {
+        //given
+        Board board = new Board();
+
+        //when
+        var result = board.getPiecePosition().get(1);
```
var를 혼용한 이유가 있나욥?

be-student replied:
테스트코드는 타입이 그렇게까지 많이 중요하다고 생각하지 않아서
너무 길어진다면 그냥 써먹으려고요
귀찮아요

drunkenhw reviewed (commented):
안녕하세요 눈우 리뷰어 박스터 입니다
정말 잘 구현해주셨더라구요~ 궁금한 것 몇가지 코멘트 남겼습니다~! :heart:

be-student reviewed (commented):
고마워 박스터

woo-chang commented on src/main/java/chess/controller/ChessController.java:
```diff
@@ -0,0 +1,97 @@
+        board = new Board();
+        printBoard();
+        while (true) {
+            RequestInfo requestInfo = repeat(() -> new RequestInfo(inputView.inputGameCommand()));
```
`repeat`라는 메서드명을 처음 보았을 때 반복을 왜 하지라는 의문이 조금 들었습니다!

반복에 집중하기보다는 유효한 `RequestInfo`를 만들기 위함을 메서드명에 드러내 보는건 어떨까요?

be-student replied:
좋아요...  지금 구조가 되게 애매한 것 같아요
감사합니다

woo-chang commented on src/main/java/chess/controller/ChessController.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private void printBoard() {
+        outputView.printBoard(board.getPiecePosition());
```
`outputView.printBoard`로도 의미를 알 수 있을 것 같은데 메서드 분리하지 않고 그대로 사용하는건 어떨까요?

be-student replied:
좋은데요? 그렇게 바꿔볼게요

woo-chang commented on src/main/java/chess/controller/ChessController.java:
```diff
@@ -0,0 +1,97 @@
+        }
+    }
+
+    private static class RequestInfo {
```
컨트롤러에서만 사용되는 클래스를 다음과 같이 관리하는거 좋은데요 👍

be-student replied:
컨트롤러에서만 사용되어서 이렇게 관리하려고 했는데, 리뷰어가 view 에서 사용하는 dto 로 패키지를 분리하는 쪽이 더 깔끔해보인다고 해서 그렇게 해보려고요
ㅋㅋㅋ 감사합니다

woo-chang commented on src/main/java/chess/domain/Color.java:
```diff
@@ -0,0 +1,17 @@
+public enum Color {
+    BLACK(-1),
+    WHITE(1),
+    NONE(0);
```
`EMPTY`를 두면서 `NONE`, `EMPTYSTATE`와 같은 추가적인 데이터가 늘어나는 느낌인데 그럼에도 `EMPTY`가 꼭 필요한 이유가 궁금합니다!

be-student replied:
null 을 도메인 내부에서 안 쓰이게 하려고 했는데요
도메인 내부에서 null 이 돌아다니게 된다면 이건 어디서부터 어디까지 null check 를 해야하지? 라는 생각을 갖게 되는 것 같아요
null 이 아얘 안돌아 다니면 도메인 내부에서는 적어도 null 체크를 안 해도 된다는 장점이 있어보였어요

woo-chang commented on src/main/java/chess/domain/File.java:
```diff
@@ -0,0 +1,57 @@
+    E("e", 5),
+    F("f", 6),
+    G("g", 7),
+    H("h", 8);
```
`String`을 사용하지 않고 `Character`로 두면 `order` 필드가 필요없어질 것 같아요 :)

be-student replied:
좋은데요? char 로 중간에 되게 많이 바꾸고 싶었어요

woo-chang commented on src/main/java/chess/domain/Piece.java:
```diff
@@ -0,0 +1,47 @@
+
+    public Piece(String type, Color color) {
+        this(PieceType.from(type), color);
+    }
```
생성자 2개다 `public`으로 사용하시는걸까요? 🤔

be-student replied:
그러게요.....? 저게 뭐죠? 누가 저렇게 짠거죠?

woo-chang commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+        this.rank = rank;
+    }
+
+    public static Position of(char fileName, char rankName) {
+        return of(String.valueOf(fileName), String.valueOf(rankName));
+    }
+
+    private static Position of(String fileName, String rankName) {
+        File file = File.from(fileName);
+        Rank rank = Rank.from(rankName);
+        Map<Rank, Position> rankToPosition = cache.get(file);
+        return rankToPosition.get(rank);
+    }
```
```java
private static Position of(char fileName, char rankName) {
    return of(File.from(fileName), Rank.from(rankName));
}
```

`File`과 `Rank` 필드를 `char`로 두고 아래의 메서드도 활용하면 위와 같이 간소화할 수 있을 것 같습니다

be-student replied:
아주 좋아요 char 를 쓰면서 바뀔 것 같아보여요 너무 좋은데요?

woo-chang commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+                '}';
+    }
+
+    public int getRankDifference(Position other) {
+        return rank.getDifference(other.rank);
+    }
+
+    public int getFileDifference(Position other) {
+        return file.getDifference(other.file);
+    }
```
여기도 메서드 분리하지 않고 바로 사용하는건 어떻게 생각하시나요?

be-student replied:
1줄짜리 메서드가 단순히 호출만 하는 경우이고, 메서드 명도 적절하지 않았던 느낌이 드네요 좋아요

woo-chang commented on src/main/java/chess/domain/Rank.java:
```diff
@@ -0,0 +1,56 @@
+    EIGHT("8", 8);
+    private static final int SKIP_FIRST = 1;
+
+    private final String rankName;
+    private final int order;
```
`ditto`: `char`이였으면 어땠을까용

be-student replied:
바꾸겠습니다! ㅋㅋㅋㅋ

woo-chang commented on src/main/java/chess/domain/state/InitialPawnState.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    private boolean isOneDiagonalMove(int x, int y) {
+        return Math.abs(x) == 1 && y == 1;
```
`y`가 `rank`를 의미하는거면 검은색 `Pawn`은 위에서 내려오기에 `y`가 `-1`인 경우도 존재하지 않나요?

be-student replied:
Piece 쪽을 보면, 
```java
if (!moveState.canMove(x, color.colorDirection(y), piece.compareColor(color))) {
    throw new IllegalPieceMoveException();
}
```
같은 형태로 되어있는데요
여기서 -1 로 가는 것을 색깔에 맞게 바꿔주기에 괜찮을 것 같아보여요

물론 blackPawn 을 따로 둘 수도 있겠지만, 바깥에 있는 것이 더 깔끔해보이는 것 같아요

woo-chang commented on src/main/java/chess/domain/state/MovedPawnState.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    private boolean isOneDiagonalMove(int x, int y) {
+        return Math.abs(x) == 1 && y == 1;
```
`ditto`: `y`가 `-1`인 경우가 있지 않나요?

be-student replied:
ditto

woo-chang commented on src/main/java/chess/domain/state/QueenState.java:
```diff
@@ -0,0 +1,34 @@
+        return Math.abs(x) == Math.abs(y);
+    }
+
+    private boolean isValidStraightRequest(int x, int y) {
+        if (x == 0 && y != 0) {
+            return true;
+        }
+        return x != 0 && y == 0;
+    }
```
`return (x == 0 && Math.abs(y) > 0) || (y == 0 && Math.abs(x) > 0)` 요런 방법은 어떨까요?

be-student replied:
지금도 사실 2개를 합칠 수 있는데, 애매해보여서 따로 뒀는데요
중간에 || 문이 있는 부분이 가독성이 살짝 애매해보였어요

woo-chang commented on src/test/java/chess/domain/PositionFixture.java:
```diff
@@ -0,0 +1,24 @@
+
+public class PositionFixture {
+
+    public static final Position A1 = Position.of(File.A, Rank.ONE);
+    public static final Position A2 = Position.of(File.A, Rank.TWO);
+    public static final Position A3 = Position.of(File.A, Rank.THREE);
+    public static final Position A4 = Position.of(File.A, Rank.FOUR);
+    public static final Position A5 = Position.of(File.A, Rank.FIVE);
+    public static final Position A7 = Position.of(File.A, Rank.SEVEN);
+    public static final Position B2 = Position.of(File.B, Rank.TWO);
+    public static final Position B3 = Position.of(File.B, Rank.THREE);
+    public static final Position B4 = Position.of(File.B, Rank.FOUR);
+    public static final Position C1 = Position.of(File.C, Rank.ONE);
+    public static final Position C3 = Position.of(File.C, Rank.THREE);
+    public static final Position D1 = Position.of(File.D, Rank.ONE);
+    public static final Position D4 = Position.of(File.D, Rank.FOUR);
+    public static final Position F1 = Position.of(File.F, Rank.ONE);
+    public static final Position E1 = Position.of(File.E, Rank.ONE);
+    public static final Position E2 = Position.of(File.E, Rank.TWO);
+    public static final Position E3 = Position.of(File.E, Rank.THREE);
+    public static final Position E4 = Position.of(File.E, Rank.FOUR);
+    public static final Position E5 = Position.of(File.E, Rank.FIVE);
+}
```
테스트 픽스처 너무 좋은데요 👍

woo-chang reviewed (commented):
수고하셨습니다 :) 많이 배워갑니다 누누 고수 👾

Choi-JJunho commented on src/main/java/chess/domain/Board.java:
```diff
@@ -0,0 +1,104 @@
+
+    private void makePiecesExceptPawns(Color color, Rank rank) {
+        List<PieceType> highPieceOrder = orderedPiece();
+        for (int i = 0; i < RANK_SIZE; i++) {
+            Position position = Position.of(File.from(i + 1), rank);
+            Piece piece = new Piece(highPieceOrder.get(i), color);
+            piecePosition.put(position, piece);
+        }
```
RANK_SIZE를 Rank에게 주는건 어떤가요?

be-student replied:
여기에서만 사용되기에, RANK_SIZE 를 하는 것으로 한건데요
생각해보면 여기서 관리할게 아니었네요
죄송합니다

Choi-JJunho commented on src/main/java/chess/domain/File.java:
```diff
@@ -0,0 +1,57 @@
+        return other.order - order;
+    }
+
+    public List<File> createPath(File other) {
+        List<File> files = IntStream.range(Math.min(order, other.order), Math.max(order, other.order))
+                .skip(SKIP_FIRST)
+                .mapToObj(File::from)
+                .collect(Collectors.toList());
+        if (order > other.order) {
+            Collections.reverse(files);
+        }
+        return files;
+    }
```
좋은데요?

be-student replied:
허브 복붙 감사합니다

Choi-JJunho commented on src/main/java/chess/domain/PieceType.java:
```diff
@@ -0,0 +1,54 @@
+    KING("k", List.of(KingState.getInstance())),
+    PAWN("p", List.of(InitialPawnState.getInstance(), MovedPawnState.getInstance())),
+    EMPTY(".", List.of(EmptyState.getInstance()));
+    private static final int INITIAL_STATE = 0;
```
개행 해주면 안되나요?

be-student replied:
여기 개행이 필요한가요? 애매하네요
일단 바꿔는 놓을게요

Choi-JJunho commented on src/main/java/chess/domain/Position.java:
```diff
@@ -0,0 +1,120 @@
+import java.util.stream.Collectors;
+
+public class Position {
+    private static final Map<File, Map<Rank, Position>> cache = new EnumMap<>(File.class);
```
캐싱 좋은데요?

be-student replied:
좋은데요?

Choi-JJunho commented on src/main/java/chess/domain/Rank.java:
```diff
@@ -0,0 +1,56 @@
+    SIX("6", 6),
+    SEVEN("7", 7),
+    EIGHT("8", 8);
+    private static final int SKIP_FIRST = 1;
```
이거도 개행해주면 안되나요?

be-student replied:
ditto

Choi-JJunho commented on src/main/java/chess/domain/state/MoveState.java:
```diff
@@ -0,0 +1,16 @@
+        return this;
+    }
+
+    default boolean canJump() {
```
점프도 하나요? 신나네요 So Cool

be-student replied:
없어진 점프입니다... ㅠㅠ

Choi-JJunho commented on src/test/java/chess/domain/BoardTest.java:
```diff
@@ -0,0 +1,88 @@
+class BoardTest {
+
+    @Test
+    void 첫_뻔쨰_줄_테스트() {
```
첬뻔쨰쭐톄쓰트?

Shift에 고양이가 올라왔네요

be-student replied:
여기도 저희 고양이가 그랬어요
저는 모르는 일이었어요

Choi-JJunho commented on src/test/java/chess/domain/BoardTest.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    @Test
+    void 두_뻔쨰_줄_테스트() {
```
쨰? ditto

be-student replied:
매섭네요 진짜 
저희 고양이가 그랬어요

Choi-JJunho commented on src/test/java/chess/domain/ColorTest.java:
```diff
@@ -0,0 +1,20 @@
+@SuppressWarnings({"NonAsciiCharacters", "SpellCheckingInspection"})
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+class ColorTest {
+    @ParameterizedTest
```
개행 해주세요

be-student replied:
개행 했습니다!

Choi-JJunho reviewed (commented):
누누 안녕하세요 리뷰어 주노입니다~
이미 이상한 사람이 베로와 포이의 이름을 빌려가며 좋은 코멘트를 많이 남겨주셨네요

컨벤션 위주로 코멘트 남겼는데 확인해주세요~

be-student committed:
refactor: controller 에서 도메인 객체를 response 로 변경하도록 수정

be-student committed:
refactor: Position 에서 직선 경로가 아니면 빈 List 를 반환한다

be-student committed:
fix: 필요하지 않은 테스트 disabled

be-student committed:
refactor: rankSize 를 rank 에서 관리하도록 변경

be-student committed:
fix: 오타 수정

be-student committed:
style: 줄바꿈 변경

be-student committed:
feat: pieceType 의 필드명 name 으로 수정

be-student committed:
typo: 오타 수정

be-student committed:
refactor: x,y 변수명 fileDifference, rankDifference 로 변경

be-student committed:
chore: ArrayList 타입 List 형태로 교체

be-student committed:
chore: 안 쓰이는 생성자 제거

be-student committed:
refactor: rank, file 에 char 형으로 갖도록 변경

be-student committed:
refactor: requestInfo 클래스를 view 로 옮김

be-student committed:
refactor: piece 와, game 을 패키지로 분리

be-student committed:
refactor: piece 움직임에 대한 예외를 piece 패키지로 이동

be-student committed:
feat: game 추상화를 수준에 맞는 예외 던지기

be-student committed:
refactor: 에러 메시지에 나오는 예외 클래스명 제외

be-student committed:
feat: chessGame 에 state 패턴과, command 패턴 적용

be-student committed:
refactor: board 에서 catch 하는 부분 game 에서 catch 하도록 변경

be-student committed:
fix: 안 쓰는 파일 제거

be-student committed:
feat: 체스 위치와 관련된 상수 클래스 추가

be-student committed:
refactor: indent 2 제거

be-student committed:
chore: 안 쓰이는 코드 제거

be-student committed:
chore: 변수명 변경

be-student committed:
chore: 메서드 네이밍 변경

be-student committed:
fix: 깨진 테스트 수정

be-student committed:
refactor: 조건문에 느낌표 부분 제거

be-student committed:
fix: 캐싱 때문에, 상태가 공유되던 문제 해결

greeng00se commented on src/main/java/chess/domain/game/ChessGame.java:
```diff
@@ -0,0 +1,62 @@
+    private static final int RANK_INDEX = 1;
+    private static final int ORIGIN_INDEX = 1;
+    private static final int TARGET_INDEX = 2;
+    private static final int DESTINATION_INDEX = TARGET_INDEX;
+    private Board board;
```
한 칸 띄워주세요 nunu

greeng00se commented on src/main/java/chess/domain/game/constant/ChessPosition.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    public static Map<Position, Piece> initialPiecePositions() {
+        Map<Position, Piece> piecePositions = new HashMap<>();
+        makeBlackHighPiece(piecePositions);
+        makeBlackPawn(piecePositions);
+        makeWhitePawn(piecePositions);
+        makeWhiteHighPiece(piecePositions);
+        makeEmptyPiece(piecePositions);
+        return new HashMap<>(piecePositions);
```
만드는 부분 캐싱했다고 하지 않았어? 아직 커밋안했나보넹

greeng00se commented on src/main/java/chess/domain/game/exception/ChessGameException.java:
```diff
@@ -0,0 +1,12 @@
+package chess.domain.game.exception;
+
+public class ChessGameException extends RuntimeException {
+
+    public ChessGameException(String message) {
+        super(message);
+    }
+
+    public ChessGameException(String message, Throwable cause) {
+        super(message, cause);
+    }
```
Custom Exception 👍

greeng00se commented on src/main/java/chess/domain/piece/PieceType.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.List;
+
+public enum PieceType {
+    ROOK("r", List.of(RookState.getInstance())),
+    KNIGHT("n", List.of(KnightState.getInstance())),
+    BISHOP("b", List.of(BishopState.getInstance())),
+    QUEEN("q", List.of(QueenState.getInstance())),
+    KING("k", List.of(KingState.getInstance())),
+    PAWN("p", List.of(InitialPawnState.getInstance(), MovedPawnState.getInstance())),
+    EMPTY(".", List.of(EmptyState.getInstance()));
```
👍

greeng00se commented on src/main/java/chess/domain/piece/state/RookState.java:
```diff
@@ -0,0 +1,27 @@
+
+public class RookState implements MoveState {
+
+    private static final RookState instance = new RookState();
+
+    private RookState() {
+    }
+
+    public static RookState getInstance() {
+        return instance;
+    }
```
상태를 싱글톤으로 만든 이유가 있을까?

greeng00se commented on src/test/java/chess/domain/BoardTest.java:
```diff
@@ -0,0 +1,96 @@
+        //given
+        Board board = new Board();
+
+        //when
+        var result = board.getPieces().get(0);
+        //then
+
+        var check = result.stream()
```
```java
//when
var result = board.getPieces().get(0);

//then
var check = result.stream()
```

아래도 동일!

greeng00se commented on src/test/java/chess/domain/BoardTest.java:
```diff
@@ -0,0 +1,96 @@
+    void 같은편_말이_없는_곳으로_이동할_수_있다() {
+        //given
+        Board board = new Board();
+
+        assertDoesNotThrow(() -> {
```
`// expect` 아래도 동일!

greeng00se commented on src/test/java/chess/domain/PositionTest.java:
```diff
@@ -0,0 +1,117 @@
+        boolean result = position2 == position1;
+
+        //then
+        assertThat(result)
+                .isTrue();
```
일관성 없는 개행인 것 같습니당

greeng00se commented on src/main/java/chess/domain/game/Rank.java:
```diff
@@ -0,0 +1,57 @@
+        this.order = order;
+    }
+
+    public static Rank from(char rankName) {
+        return Arrays.stream(Rank.values())
+                .filter(it -> it.rankName == rankName)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 rank 입니다"));
+    }
+
+    public static Rank from(int rankOrder) {
+        return Arrays.stream(Rank.values())
+                .filter(it -> it.order == rankOrder)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 rank 입니다"));
+    }
```
custom과 legacy를 혼용하는 이유가 있을까?

Choi-JJunho replied:
누누 : 귀찮아서

greeng00se reviewed (commented):
누누는 페어랑 할 때 빛이 나는군요 ⭐

be-student committed:
feat: requestType 을 controller 로 이동

be-student committed:
fix: file, rank 에서도 custom 예외를 사용하도록 변경

be-student committed:
feat: position 을 String 으로 생성하는 기능 추가

be-student committed:
feat: isRunning 제거하고, 반복 가능하게 변경

be-student committed:
feat: command 를 controller 쪽으로 변경

be-student committed:
chore: 접근 제어자 변경

be-student committed:
chore: 사용되지 않는 클래스 제거

be-student committed:
feat: view 에서 사용되는 start, move, end 에 대해 컨트롤러 부분 제거

기존에는 controller 패키지에 가서, start, move, end 를 변경했어야 하지만
지금은 view 패키지에서 직접 변경할 수 있습니다

be-student committed:
feat: pieceMapper 추가

be-student committed:
feat: input 과 output 을 구분해서 의존성 방향을 설계

be-student committed:
feat: RequestType 이 바로 생성 불가능하게 변경

be-student committed:
refactor: 안 쓰는 코드 제거

be-student committed:
feat: 하위 레이어의 예외 번역

be-student committed:
refactor: pieceResponse 패키지 변경

be-student committed:
feat: 도메인에 있던 piece 이름들 controller 에서 mapper 로 변경

be-student committed:
feat: piece를 불변 객체로 변경

be-student committed:
refactor: 위치가 있는 상수 클래스 enum 으로 변경

be-student committed:
feat: 입력 재입력 과정 추가

be-student committed:
refactor: 출력하는 부분을 request type 쪽으로 옮김

be-student committed:
chore: 줄바꿈 변경

be-student committed:
fix: 정규 표현식에 ^$ 추가

be-student committed:
refactor: split 하는 부분 request 안쪽으로 이동

be-student committed:
refactor: domain 만을 보고, 실행할 수 있도록 변경
